From e8fcef902d680f947012e469e3b3523ebee8dacc Mon Sep 17 00:00:00 2001
From: Alexander Shiyan <shc_work@mail.ru>
Date: Wed, 16 Apr 2014 11:24:54 +0400
Subject: [PATCH 063/110] ARM: dts: imx51-babbage: Add USB OTG regulator node

This patch adds a regulator node and pinctrl group for USB OTG.

Signed-off-by: Alexander Shiyan <shc_work@mail.ru>
Signed-off-by: Shawn Guo <shawn.guo@freescale.com>
---
 arch/arm/boot/dts/imx51-babbage.dts | 29 ++++++++++++++++++++++++-----
 1 file changed, 24 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/imx51-babbage.dts b/arch/arm/boot/dts/imx51-babbage.dts
index a8d6168..ed9d769 100644
--- a/arch/arm/boot/dts/imx51-babbage.dts
+++ b/arch/arm/boot/dts/imx51-babbage.dts
@@ -137,17 +137,29 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 
-		reg_usb_vbus: regulator@0 {
+		reg_usbh1_vbus: regulator@0 {
 			compatible = "regulator-fixed";
 			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_usbreg>;
+			pinctrl-0 = <&pinctrl_usbh1reg>;
 			reg = <0>;
-			regulator-name = "usb_vbus";
+			regulator-name = "usbh1_vbus";
 			regulator-min-microvolt = <5000000>;
 			regulator-max-microvolt = <5000000>;
 			gpio = <&gpio2 5 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
+
+		reg_usbotg_vbus: regulator@1 {
+			compatible = "regulator-fixed";
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_usbotgreg>;
+			reg = <1>;
+			regulator-name = "usbotg_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio1 7 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
 	};
 
 	usbphy {
@@ -539,11 +551,17 @@
 			>;
 		};
 
-		pinctrl_usbreg: usbreggrp {
+		pinctrl_usbh1reg: usbh1reggrp {
 			fsl,pins = <
 				MX51_PAD_EIM_D21__GPIO2_5		0x85
 			>;
 		};
+
+		pinctrl_usbotgreg: usbotgreggrp {
+			fsl,pins = <
+				MX51_PAD_GPIO1_7__GPIO1_7		0x85
+			>;
+		};
 	};
 };
 
@@ -618,7 +636,7 @@
 &usbh1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usbh1>;
-	vbus-supply = <&reg_usb_vbus>;
+	vbus-supply = <&reg_usbh1_vbus>;
 	fsl,usbphy = <&usbh1phy>;
 	phy_type = "ulpi";
 	status = "okay";
@@ -628,5 +646,6 @@
 	dr_mode = "otg";
 	disable-over-current;
 	phy_type = "utmi_wide";
+	vbus-supply = <&reg_usbotg_vbus>;
 	status = "okay";
 };
-- 
1.9.2

